### PR TITLE
main.go: starting to write main.go

### PR DESCRIPTION
**Pull request by ahmetalpbalkan**
Some skeleton code for starting to write main executable.

### CODE REVIEW

File: cmds.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+import "github.com/Azure/azure-docker-extension/pkg/vmextension"
+
+type cmdFunc func(vmextension.HandlerEnvironment) error
+
+var (
+	cmds = map[string]struct {
+		f             cmdFunc // associated function
+		name          string  // human readable string
+		reportsStatus bool    // determines if running this should log to a .status file
```
Comment (boumenot) [inline cmds.go]: shouldReportStatus

File: cmds.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package main
+
+import "github.com/Azure/azure-docker-extension/pkg/vmextension"
+
+type cmdFunc func(vmextension.HandlerEnvironment) error
+
+var (
+	cmds = map[string]struct {
+		f             cmdFunc // associated function
```
Comment (boumenot) [inline cmds.go]: You basically defined a type, but wrapped it up in a map.  Why not just a type?

Comment (ahmetalpbalkan) [inline cmds.go]: @boumenot map is for lookup with string-->func

File: cmds.go (hunk 1 above)
Comment (ahmetalpbalkan) [inline cmds.go]: will fix.

File: cmds.go (hunk 2 above)
Comment (boumenot) [inline cmds.go]: Understood.  The compiler can check types, but it cannot do the same for strings.  If you can lean on the compiler to solve this problem I would.

**Comment by boumenot** [comment]
LGTM.

File: cmds.go (hunk 2 above)
Comment (ahmetalpbalkan) [inline cmds.go]: @boumenot I miss the full argument. the anonymous struct is still type safe. I just use the map to lookup the user-input command e.g. "enable" to the enable method below.

Comment (boumenot) [inline cmds.go]: The compiler checks this code.

```go
extension.install()
```

It does not check this code.  You have to do it.

```go
m["install"].f()
```

Comment (ahmetalpbalkan) [inline cmds.go]: fair, I'm already doing it in this case.

**Pull request merged**